pkg/operator/controller/nodeobservability/daemonset.go

```diff
@@ -32,9 +35,16 @@ func (r *NodeObservabilityReconciler) ensureDaemonSet(ctx context.Context, nodeO
 	desired := r.desiredDaemonSet(nodeObs, sa)
 	exist, current, err := r.currentDaemonSet(ctx, nameSpace)
 	if err != nil {
-		return false, nil, fmt.Errorf("failed to get DaemonSet: %v", err)
+		return false, nil, fmt.Errorf("failed to get DaemonSet: %w", err)
 	}
 	if !exist {
+		cmExists, err := r.createConfigMap(nodeObs)
+		if err != nil {
```

Comment:
I cannot put another comment on the same line but this is actually for the previous line.

`createConfigMap` is called as part of the reconciliation loop of NodeObservability CR. What if NodeObservability CR wasn't changed but the kubelet CA was rotated? You'd have to edit NodeObservability CR just to get the new version of the kubelet CA.